baseplate/thrift.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+type baseplateThriftServer struct {
+	thriftServer *thrift.TSimpleServer
+	config       ServerConfig
+	closers      []io.Closer
+	logger       log.Wrapper
+	secretsStore *secrets.Store
+}
+
+func (bts *baseplateThriftServer) Config() ServerConfig {
+	return bts.config
+}
+func (bts *baseplateThriftServer) Secrets() *secrets.Store {
+	return bts.secretsStore
+}
+
+func (bts *baseplateThriftServer) Serve() error {
+	return bts.thriftServer.Serve()
+}
+
+func (bts *baseplateThriftServer) Close() error {
+	var errors batcherror.BatchError
+	errors.Add(bts.thriftServer.Stop())
+	for _, c := range bts.closers {
+		errors.Add(c.Close())
+	}
+	return errors.Compile()
+}
+
+func initLogger(cfg ServerConfig) log.Wrapper {
+	if cfg.Log.Level == "" {
+		cfg.Log.Level = log.InfoLevel
+	}
+	level := cfg.Log.Level
+	log.InitLogger(level)
+	return log.ZapWrapper(level)
+}
+
+func initSecrets(ctx context.Context, cfg ServerConfig, logger log.Wrapper) (*secrets.Store, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*30)
+	defer cancel()
+
+	secretsStore, err := secrets.NewStore(ctx, cfg.Secrets.Path, logger)
+	if err != nil {
+		return nil, err
+	}
+	return secretsStore, nil
+}
+
+func initTracing(cfg ServerConfig, logger log.Wrapper, metrics *metricsbp.Statsd) error {
+	if err := tracing.InitGlobalTracer(tracing.TracerConfig{
+		ServiceName:      cfg.Tracing.Namespace,
+		SampleRate:       cfg.Tracing.SampleRate,
+		Logger:           logger,
+		MaxRecordTimeout: cfg.Tracing.RecordTimeout,
+		QueueName:        "main",
+	}); err != nil {
+		return err
+	}
+
+	tracing.RegisterCreateServerSpanHooks(
+		metricsbp.CreateServerSpanHook{Metrics: metrics},
+		tracing.ErrorReporterCreateServerSpanHook{},
+	)
+	return nil
+}
+
+func initSentry(cfg ServerConfig) error {
+	if err := raven.SetDSN(cfg.Sentry.DSN); err != nil {
+		return err
+	}
+	if err := raven.SetSampleRate(float32(cfg.Sentry.SampleRate)); err != nil {
+		return err
+	}
+	raven.SetEnvironment(cfg.Sentry.Environment)
+	return nil
+}
+
+func cleanup(closers []io.Closer) {
+	for _, c := range closers {
+		if err := c.Close(); err != nil {
+			log.Error("error cleaning up thrift server initialization:", err)
+		}
+	}
+}
+
+// NewBaseplateThriftServer returns a server that initializes and includes the default
+// middleware and cross-cutting concerns needed in order to be a standard Baseplate service.
+//
+// At the moment, this includes secrets management, metrics, edge contexts
+// (edgecontext.InjectThriftEdgeContext), and spans/tracing (tracing.InjectThriftServerSpan).
+func NewBaseplateThriftServer(ctx context.Context, cfg ServerConfig, processor thriftbp.BaseplateProcessor, additionalMiddlewares ...thriftbp.Middleware) (srv Server, err error) {
```

Comment:
The other rename is happening in https://github.com/reddit/baseplate.go/pull/144, so this could be renamed into `NewThriftServer`.